src/commonMain/kotlin/exchange.dydx.abacus/processor/wallet/account/AssetPositionsProcessor.kt

```diff
@@ -2,18 +2,60 @@ package exchange.dydx.abacus.processor.wallet.account
 
 import exchange.dydx.abacus.processor.base.BaseProcessor
 import exchange.dydx.abacus.protocols.ParserProtocol
+import exchange.dydx.abacus.state.internalstate.InternalAssetPositionState
 import exchange.dydx.abacus.utils.mutable
 import exchange.dydx.abacus.utils.safeSet
+import indexer.codegen.IndexerAssetPositionResponseObject
 
-internal class AssetPositionsProcessor(parser: ParserProtocol) : BaseProcessor(parser) {
-    private val itemProcessor = AssetPositionProcessor(parser = parser)
+internal class AssetPositionsProcessor(
+    parser: ParserProtocol,
+    private val itemProcessor: AssetPositionProcessorProtocol = AssetPositionProcessor(parser = parser)
+) : BaseProcessor(parser) {
 
-    internal fun received(payload: Map<String, Any>?): Map<String, Any>? {
+    // From REST call
```

Comment:
[suggestion] since we have the opportunity to make new names, can we make them descriptive of the source that's calling them? 
e.g.:
processRESTCall

vs

processWebsocketMessage

or something

Reply:
Yeah.. Let's do the naming change afterwards...  There are some complexities involving the current tests code (where it processes the REST function whereas in the real app the data is actually from Websocket) that we will have to clean up.

Reply:
got it, sounds good! can we add a ticket to clean up and put as a comment?